### ISSUE TITLE
Some issues/bugs/suggestions

### ISSUE THREAD

**Issue by N1GHT-090** [CriticalFlaw/flawhud#185]
Hello, and thank you for the hud. I think it is one of nicest available. First sorry for such a long post. Im not a programmer but I like to tinker with hud files for fun and figured out most of this stuff with vgui_drawtree and trial and error. I didnt know if i should make seperate issuses for each thing, I didnt want to clutter up your issue tracker. I used the hud editor to apply some of the alternative settings so maybe you are not working on those features at the moment and there may be reasons why some of the things I have found are the way they are.

The first bug I have found deals with the "Health Style - Colored Text" option in the editor. With pyro or heavy when you run your ammo low and the ammo text turns red, going to resupply or picking up ammo doesnt reset the color to white, instead it just stays red. I seemed to have fixed this by editing line 81 in the hudanimations_custom.txt file. The edit I made was to change the animation `Animate AmmoInClip` to `Animate AmmoNoClip` . It seems like maybe you intended this to be "NoClip" instead of "InClip" as in the section above on line 65 it is `Animate AmmoNoClip` 

The second bug is with "HUD Style - Alternate". The voice popup when people are speaking overlaps the meters on the right side like heavy's food meter. I fixed this by changing the ypos of `HudVoiceStatus` in hudlayout.res. I set it to -50 but that is just personal preference on my part and may be too much. I dont know what the best number would actually be.

The third issue has to do with the halloween backgrounds for Scream Fortress. With "Main Menu Class Images" enabled in the editor sometimes the characters randomly dont show up. This is only for Scream Fortress. I looked in the mainmenuoverride.res file and saw you only have the Scream Fortress 2017 background enabled. That may be your personal preference however, I dont know. It seems TF2 is still doing something in the background though, Im pretty sure I have discovered that two default halloween backgrounds seem to disable characters to avoid covering something on the background. They are Halloween 2014 and Halloween 2015 backgrounds, found here https://wiki.teamfortress.com/wiki/Category:Menu_backgrounds . It seems TF2 is still rotating background options behind the scenes even if you only have enabled and only see the 2017 Scream Fortress background and it still disables characters when the 2014/2015 backgrounds would normally be shown.

I found by looking in ahud that you can fix the seemingly "random" missing menu characters by enabling the other backgrounds. That way when the game disables the characters it shows the two backgrounds designed for that. But you may have chosen to only have the Scream Fortress 2017 background enabled by design if you dont use the the character models. You wouldn't notice this unless they are enabled. Here are the lines from ahud enabling the other backgrounds if you need them.
```
		"if_halloween_0"
		{
			"image"	"../console/title_team_halloween2011_widescreen"
		}
		"if_halloween_1"
		{
			"image"	"../console/title_team_halloween2012_widescreen"
		}
		"if_halloween_2"
		{
			"image"	"../console/title_team_halloween2013_widescreen"
		}
		"if_halloween_3"
		{
			"image"	"../console/title_team_halloween2014_widescreen"
		}
		"if_halloween_4"
		{
			"image"	"../console/title_team_halloween2015_widescreen"
		}
		"if_halloween_5"
		{
			"image"	"../console/title_scream_fortress_2017_widescreen"
		}
```
The last thing is just a suggestion. The size of the default damage numbers are a little large for me personally, so I change them to size 14 in the clientscheme_fonts.res and huddamageaccount.res files. Maybe you could add a font size option in the editor for it, however that may be too much work for a such small option.

 Thanks again for the hud.

**Comment by CriticalFlaw** [comment]
Thank you for your report. I've looked through what you have written and put my notes below.

You're right about the first issue, there's a duplicate line for `AmmoInClip` in place of `AmmoNoClip`. You're welcome to submit a pull request fixing this (and other) issues - if you're not sure how to, you can let me know then I'll fix them and credit you for reporting them.

The placement of the voice indicator is an oversight in the hud editor file, I'll add it in - likely positioning it over the player model to make it look like they are talking - it seems like the most natural place for it.

That's a really good catch on the halloween backgrounds. I had only restricted it to the one flawhud uses now because all other backgrounds are much busier and do not fit well with the other main menu elements. Ideally I'd like the halloween background to always appear during halloween, with or without the character images.

Damage numbers are one of those subjective things where you're damned if you change it and damned if you do. So best to leave it to the user to change in the editor rather than change the current damage size for everyone. I'll make a note of adding that option to the editor. Cheers.

**Comment by N1GHT-090** [comment]
Thanks for the reply. I don't know how to submit fixes or pull requests so you can do that. Like I said I am not a programmer just a tinkerer. Ive been fixing some of this on my own version but felt I should probably just make a github account and let you know about the stuff that seemed to be actually broken and not just my preference. Also I was worried someone else was encountering it and didn't know how to fix it and maybe would stop using the hud.

I figured as much with the halloween background, it doesn't really break anything just kind of confusing, as the character seems to disappear randomly. I thought my game was bugged at first. I don't know if its possible to fix and keep just the 2017 background, that's above my pay grade lol. If you keep it as is you could maybe make a note of it on the main page for people who get confused as there are probably more people playing this time of year, but again it seems like this will only be an issue for Scream Fortress so maybe not necessary. Again thanks and keep up the great work.

**Comment by CriticalFlaw** [comment]
Fixed the noted issues ec48e693c9317d89aa7c28414a66abd5f07ac73b and updated the editor schema in https://github.com/CriticalFlaw/TF2HUD.Editor/commit/1e98fc147b3602dbb036c79607186c7d70f3fffd to fix the placement of the voice indicator along with few other elements when the Alternate layout style is enabled. Thank you again for reporting these issues.
